src/backend/joanie/edx_imports/api/__init__.py

```diff
@@ -0,0 +1,45 @@
+"""Viewset for course_run in the edx_imports app."""
+
+from http import HTTPStatus
+
+from rest_framework.decorators import (
+    api_view,
+    authentication_classes,
+    permission_classes,
+)
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from joanie.core.models import CourseRun
+from joanie.core.permissions import HasAPIKey
+from joanie.core.serializers import AdminCourseRunSerializer
+
+
+@api_view(["GET"])
+@permission_classes([HasAPIKey])
+@authentication_classes([])
+def course_run_view(request: Request):
+    """
+    Look for course_runs filter by a resource_link.
+    """
+    resource_link = request.query_params.get("resource_link")
+    if resource_link is None:
+        return Response(
+            {"detail": "Query parameter `resource_link` is required."},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    course_run = (
+        CourseRun.objects.filter(resource_link__icontains=resource_link)
+        .select_related("course")
+        .first()
+    )
+
+    if not course_run:
+        return Response(
+            {"detail": "Course run not found."},
+            status=HTTPStatus.NOT_FOUND,
+        )
+
+    serializer = AdminCourseRunSerializer(course_run)
```

Comment:
Is it the sort of context where we want to make sure that the serialized data is valid ? 

`serializer.is_valid(raise_exception=True)`

Reply:
The data is coming from the database. We validate it when it comes from a request. Here the data is already validates.